packages/mattermost-redux/src/reducers/entities/apps.ts

```diff
@@ -13,15 +13,50 @@ import {validateBindings} from 'mattermost-redux/utils/apps';
 
 export function bindings(state: AppBinding[] = [], action: GenericAction): AppBinding[] {
     switch (action.type) {
+    case AppsTypes.FAILED_TO_FETCH_APP_BINDINGS: {
+        if (!state.length) {
+            return state;
+        }
+
+        return [];
+    }
     case AppsTypes.RECEIVED_APP_BINDINGS: {
         validateBindings(action.data);
         return action.data || [];
     }
+    case AppsTypes.APPS_PLUGIN_DISABLED: {
+        if (!state.length) {
+            return state;
+        }
+
+        return [];
+    }
+    default:
+        return state;
+    }
+}
+
+export function pluginEnabled(state = true, action: GenericAction): boolean {
+    switch (action.type) {
+    case AppsTypes.APPS_PLUGIN_ENABLED: {
+        return true;
+    }
+    case AppsTypes.APPS_PLUGIN_DISABLED: {
+        return false;
+    }
+    case AppsTypes.RECEIVED_APP_BINDINGS: {
+        return true;
+    }
+    case AppsTypes.FAILED_TO_FETCH_APP_BINDINGS: {
+        return false;
```

Comment:
@larkox We may want to check if we received a 404 here, as there may be another reason why this failed. If the Apps plugin crashes, we will later receive a "plugin enabled" event to then know that it's available, so in that case it's better not to check.

The case we would want to avoid is there being a problem with the Apps plugin that causes it to return a non-404 error, and has the problem corrected without the plugin restarting, because then the frontend is left in a state where it thinks the plugin is unavailable, even though it is available after the problem is corrected.

Reply:
More than a problem with the Apps plugin, what I foresee are problems either with the server (server is down) or with the client (client connection is lost).

We could tie fetching the bindings to the websocket reconnection (not sure if we already do), and therefore updating the apps enabled value on reconnection.

Reply:
@larkox Working on this task for mobile, I'm actually thinking now that we shouldn't invalidate the bindings for the "plugin disable" websocket event. If the user is looking at a thread on mobile, and the Apps plugin is toggled, the thread bindings would be cleared and wouldn't appear again right?

Reply:
If we clear everything, the binding will dissappear, which is expected and desired, since we don't want the user to click on a binding that does not work. If the plugin gets enabled again, we could consider doing some logic to guess if we are in a thread, and bring the bindings, but that is not so critical (the end user does not need to have the bindings the instant the plugin gets online, and all users asking for the bindings at the same time could be really bad for the server).